x-pack/legacy/plugins/ml/public/services/new_job_capabilities._service.test.ts

```diff
@@ -19,9 +18,8 @@ jest.mock('./ml_api_service', () => ({
   },
 }));
 
-const indexPattern: IndexPatternWithType = {
+const indexPattern: any = {
```

Comment:
nit - can't we use the `IndexPattern` type here?

Reply:
For this to be a real `IndexPattern` it'd have to have about 40 odd properties, so i went with `any` as `initializeFromIndexPattern` only needs `title` and `type`.
I considered casting it to `IndexPattern` but I felt that was more hacky and the test didn't really gain much.

Reply:
In the recent upstream PR for `IndexPattern` they use `(... as unknown) as IndexPattern` - would that be something we can use here too?

Reply:
good point about using `unknown` like that. i'll change this to be consistent.